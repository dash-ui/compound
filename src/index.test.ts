import compound from './index'
import {styles} from '@dash-ui/styles'

describe('compound', () => {
  afterEach(() => {
    document.getElementsByTagName('html')[0].innerHTML = ''
    styles.dash.sheet.flush()
    styles.dash.cache.clear()
  })

  it('should create a compound style', () => {
    const compoundStyles = compound(styles)
    const text = compoundStyles({
      default: styles.one({
        backgroundColor: 'white',
      }),
      color: styles({
        red: {color: 'red'},
      }),
      size: styles.lazy((fontSize: string) => ({
        fontSize,
      })),
    })

    expect(text.css({color: 'red', size: '1rem'})).toBe(
      'background-color:white;color:red;font-size:1rem;'
    )
    text({color: 'red', size: '1rem'})
    expect(document.querySelectorAll(`style[data-dash]`).length).toBe(1)
    expect(document.querySelectorAll(`style[data-dash]`)[0]).toMatchSnapshot()

    // Cached
    expect(text.css({color: 'red', size: '1rem'})).toBe(
      'background-color:white;color:red;font-size:1rem;'
    )
    text({color: 'red', size: '1rem'})
    expect(document.querySelectorAll(`style[data-dash]`).length).toBe(1)
    expect(document.querySelectorAll(`style[data-dash]`)[0]).toMatchSnapshot()
  })
})
