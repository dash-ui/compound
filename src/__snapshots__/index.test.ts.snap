// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compound should create a compound style 1`] = `".ui-1ke32n{background-color:white;color:red;font-size:1rem;}"`;

exports[`compound should create a compound style 2`] = `".ui-1ke32n{background-color:white;color:red;font-size:1rem;}"`;

exports[`compound should create an atomic compound style 1`] = `".ui-12krds1{background-color:white;}"`;

exports[`compound should create an atomic compound style 2`] = `".ui-9rfate{color:red;}"`;

exports[`compound should create an atomic compound style 3`] = `".ui-1q19p6e{font-size:1rem;}"`;

exports[`compound should create an atomic compound style 4`] = `".ui-12krds1{background-color:white;}"`;

exports[`compound should create an atomic compound style 5`] = `".ui-9rfate{color:red;}"`;

exports[`compound should create an atomic compound style 6`] = `".ui-1q19p6e{font-size:1rem;}"`;

exports[`compound should create an atomic compound style ad hoc 1`] = `".ui-12krds1{background-color:white;}"`;

exports[`compound should create an atomic compound style ad hoc 2`] = `".ui-9rfate{color:red;}"`;

exports[`compound should create an atomic compound style ad hoc 3`] = `".ui-1q19p6e{font-size:1rem;}"`;

exports[`compound should create an atomic compound style ad hoc 4`] = `".ui-12krds1{background-color:white;}"`;

exports[`compound should create an atomic compound style ad hoc 5`] = `".ui-9rfate{color:red;}"`;

exports[`compound should create an atomic compound style ad hoc 6`] = `".ui-1q19p6e{font-size:1rem;}"`;
